sdk/vaa/structs.go

```diff
@@ -642,6 +642,19 @@ func (v *BatchVAA) ObsvHashArray() []common.Hash {
 	return hashes
 }
 
+// Verify Signature checks that the provided address matches the address that created the signature for the provided data
+func VerifySignature(data []byte, signature *Signature, address common.Address) bool {
```

Comment:
minor nit: I suggest renaming this parameter from `data` -> `digest` because it's actually supposed to be the output of `vaa.SigningMsg(data).Bytes()` and not the raw data directly.